themes/10up-block-theme/assets/css/base/link.css

```diff
@@ -0,0 +1,11 @@
+/*
+ * Ensure that we are not overriding the editor styles of the
+ * WordPress Components.
+ *
+ * This is a workaround for: https://github.com/WordPress/gutenberg/issues/10178
+ *
+ * using :where to prevent the specificity increase of using :not
+ */
+a:where(:not(.components-external-link, :has(> img:only-child), .wp-element-button)) {
```

Comment:
I would like to see a similar file for form elements.  I've seen a few projects that don't scope their input/select etc selectors and it affects the editor placeholder components. I've used the following with success to prevent bleeding into those. 

```css
// base/forms.css
&:where(:not(.components-placeholder)) {

    input {...}
}
```